gradle-plugin/src/gradleTest/java/com/squareup/anvil/plugin/ClasspathTest.kt

```diff
@@ -0,0 +1,146 @@
+import com.rickbusarow.kase.gradle.versions
+import com.rickbusarow.kase.kase
+import com.squareup.anvil.plugin.buildProperties.anvilVersion
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.api.TestFactory
+
+internal class BuildType<T : GradleKotlinTestVersions>(
+  override val displayName: String,
+  val runtimeName: String,
+  val compileName: String,
+  val params: List<T>,
+  val setup: (GradleKotlinTestVersions) -> String,
+) : Kase4<String, String, List<T>, (GradleKotlinTestVersions) -> String> by kase(
+  a1 = runtimeName,
+  a2 = compileName,
+  a4 = setup,
+  a3 = params,
+)
+
+internal class ClasspathTest : BaseGradleTest() {
+
+  val kotlinJvm = BuildType(
+    displayName = "kotlin-jvm",
+    runtimeName = "runtimeClasspath",
+    compileName = "compileClasspath",
+    params = kaseMatrix.versions(GradleKotlinTestVersions),
+  ) { versions ->
+
+    """
+    plugins {
+      kotlin("jvm") version "${versions.kotlinVersion}"
+      id("com.squareup.anvil") version "$anvilVersion"
+    }
+    """.trimIndent()
+  }
+
+  val androidApp = BuildType(
+    displayName = "android-app",
+    runtimeName = "debugRuntimeClasspath",
+    compileName = "debugCompileClasspath",
+    params = kaseMatrix.versions(GradleKotlinAgpTestVersions),
+  ) { versions ->
+    versions as GradleKotlinAgpTestVersions
+    """
+    plugins {
+      id("com.android.application") version "${versions.agpVersion}"
+      kotlin("android") version "${versions.kotlinVersion}"
+      id("com.squareup.anvil") version "$anvilVersion"
+    }
+    
+    ${androidBlock()}
+    """.trimIndent()
+  }
+  val androidLibrary = BuildType(
+    displayName = "android-library",
+    runtimeName = "debugRuntimeClasspath",
+    compileName = "debugCompileClasspath",
+    params = kaseMatrix.versions(GradleKotlinAgpTestVersions),
+  ) { versions ->
+    versions as GradleKotlinAgpTestVersions
+    """
+    plugins {
+      id("com.android.library") version "${versions.agpVersion}"
+      kotlin("android") version "${versions.kotlinVersion}"
+      id("com.squareup.anvil") version "$anvilVersion"
+    }
+    
+    ${androidBlock()}
+    """.trimIndent()
+  }
+
+  /**
+   * Why can't we just use dependency-guard to check the classpath?
+   *
+   * Dependency-Guard can check the `runtimeClasspath` and `compileClasspath` configurations
+   * of the `gradle-plugin` project. That tells us what Anvil will contribute to a target project's
+   * `classpath` configuration (the one modified in the `buildScript` block).
+   *
+   * For this test, we're testing what dependencies Anvil adds to the target project's own
+   * `runtimeClasspath` and `compileClasspath` configurations.  Specifically, it adds the Anvil
+   * annotations dependency -- but is it added correctly as a `compileOnly` dependency,
+   * or is it being added as `implementation` or something else?
+   */
+  @TestFactory
+  fun `annotations aren't in the runtime classpath`() = listOf(
```

Comment:
Nit: Since we also care about verifying that the annotations are included in the compile classpath, something like `annotations are only included in the compile classpath` might be better for encoding the expectations

Reply:
That's an excellent point.  Thanks!